server/src/main/java/org/opensearch/search/internal/ShardSearchRequest.java

```diff
@@ -243,8 +243,6 @@ public ShardSearchRequest(StreamInput in) throws IOException {
         clusterAlias = in.readOptionalString();
         if (in.getVersion().onOrAfter(LegacyESVersion.V_7_0_0)) {
             allowPartialSearchResults = in.readBoolean();
-        } else if (in.getVersion().onOrAfter(LegacyESVersion.V_6_3_0)) {
```

Comment:
@nknize I think we have an issue here,  the `in.readOptionalBoolean();` should be followed for the version prior  `LegacyESVersion.V_7_0_0`, it looks like we have to do use `V_6_4_0` instead of removing the condition altogether: 

```
} else if (in.getVersion().onOrAfter(LegacyESVersion.V_6_4_0)) {
    allowPartialSearchResults = in.readOptionalBoolean();
}
```

WDYT?

Reply:
That logic path won't ever be executed.

`minIndexCompatVersion` for OpenSearch 2.0.0 is ES 7.0.0 and `minCompatVersion` is ES 7.10.0 so a Mixed cluster of OpenSearch 2.0.0 and ES 6.x nodes or indexes is not possible. The first `if` condition in this version will always be `true`. 

This PR is the next in a line of incremental "remove all 6.x logic" PRs.